bugsnag-android-performance/src/main/kotlin/com/bugsnag/android/performance/internal/SpanFactory.kt

```diff
@@ -275,16 +279,20 @@ public class SpanFactory(
             isFirstClass == true
         }
 
-        return if (localRenderingMetricsSource != null || localCpuMetricsSource != null) {
-            SpanMetricsSnapshot(localRenderingMetricsSource, localCpuMetricsSource)
+        val localMemoryMetricsSource = memoryMetricsSource?.takeIf {
+            isFirstClass == true
+        }
+
+        return if (localRenderingMetricsSource != null || localCpuMetricsSource != null || localMemoryMetricsSource != null) {
+            SpanMetricsSnapshot(localRenderingMetricsSource, localCpuMetricsSource, localMemoryMetricsSource)
         } else {
             null
         }
     }
 
     private fun UUID.isValidTraceId() = mostSignificantBits != 0L || leastSignificantBits != 0L
 
-    private companion object {
+    internal companion object {
         const val ONE_SECOND = 1000L
```

Comment:
This value should probably be extract to somewhere more common - rather than being a member of the `SpanFactory` companion. Possibly move it to the `AbstractSampledMetricsSource` companion and rename it `SAMPLE_DELAY_ONE_SECOND`